src/_png.cpp

```diff
@@ -287,10 +295,23 @@ static PyObject *_read_png(PyObject *filein, bool float_result)
         py_file = filein;
     }
 
-    if ((fp = mpl_PyFile_Dup(py_file, (char *)"rb", &offset))) {
+    #if PY3K
+    if (close_file) {
+    #else
+    if (close_file || PyFile_Check(py_file)) {
+    #endif
+        fp = mpl_PyFile_Dup(py_file, (char *)"rb", &offset);
+    }
```

Comment:
There probably will be a conflict fist time we do a merge to master that needs manual resolving but after that Git will mark the commit as already merged and there is no issue